tests/testthat/test-bar_plot_emission_profile.R

```diff
@@ -1,30 +1,86 @@
+  expected_risk_categories <- risk_category_levels()
+  expect_true(setequal(risk_categories, expected_risk_categories))
+})
+
+test_that("returns correct risk category values for worst_case mode", {
+  data <- example_without_financial(!!aka("risk_category") := risk_category_levels())
+  data <- prepare_bar_plot_emission_profile(data, benchmarks(), "worst_case")
+  risk_categories <- levels(data$risk_category_var)
+  expected_risk_categories <- risk_category_levels()
+  expect_true(setequal(risk_categories, expected_risk_categories))
+})
+
+test_that("returns correct benchmarks values for equal weight mode", {
+  data <- example_without_financial()
+  data <- prepare_bar_plot_emission_profile(data, benchmarks(), "equal_weight")
+  benchmarks <- unique(data$benchmark)
+  expected_benchmarks <- example_without_financial() |>
+    pull(benchmark) |>
+    unique()
+  expect_true(all(benchmarks %in% expected_benchmarks))
+})
+
+test_that("returns correct benchmarks values for best case mode", {
   data <- example_without_financial()
-  plot <- bar_plot_emission_profile(data, benchmarks())
-  benchmarks <- unique(plot |> plot_data("benchmark"))
-  expected_benchmarks <- data |>
+  data <- prepare_bar_plot_emission_profile(data, benchmarks(), "best_case")
+  benchmarks <- unique(data$benchmark)
+  expected_benchmarks <- example_without_financial() |>
     pull(benchmark) |>
     unique()
   expect_true(all(benchmarks %in% expected_benchmarks))
 })
 
-test_that("calculated proportions are less or equal to 1", {
+test_that("returns correct benchmarks values for worst case mode", {
   data <- example_without_financial()
-  plot <- bar_plot_emission_profile(data, benchmarks())
-  proportions <- plot |> plot_data("proportion")
-  expect_true(proportions >= 0 & proportions <= 1)
+  data <- prepare_bar_plot_emission_profile(data, benchmarks(), "worst_case")
+  benchmarks <- unique(data$benchmark)
+  expected_benchmarks <- example_without_financial() |>
+    pull(benchmark) |>
+    unique()
+  expect_true(all(benchmarks %in% expected_benchmarks))
+})
+
+test_that("proportions are less or equal to 1 for equal weight mode", {
+  data <- example_without_financial()
+  data <- data |>
+    prepare_bar_plot_emission_profile(benchmarks(), mode = "equal_weight")
+
+  proportions <- data$proportion
+  expect_true(all(proportions >= 0 & proportions <= 1))
+})
+
+test_that("proportions are less or equal to 1 for best case mode", {
+  data <- example_without_financial()
+  data <- data |>
+    prepare_bar_plot_emission_profile(benchmarks(), mode = "best_case")
+
+  proportions <- data$proportion
+  expect_true(all(proportions >= 0 & proportions <= 1))
+})
+
+test_that("proportions are less or equal to 1 for worst case mode", {
+  data <- example_without_financial()
+  data <- data |>
+    prepare_bar_plot_emission_profile(benchmarks(), mode = "worst_case")
+
+  proportions <- data$proportion
+  expect_true(all(proportions >= 0 & proportions <= 1))
 })
```

Comment:
Nice! The `prepare*()` functions now belong to a different module and is easier to test than the visuals. That module can stay here or if you want to enforce stronger boundaries then we can move it to a separate package, e.g. tilltPlotPrepare.